client/state/themes/themes-list/reducer.js

```diff
@@ -103,7 +105,7 @@ export default ( state = initialState, action ) => {
 		case DESERIALIZE:
 			return initialState;
 		case SERVER_DESERIALIZE:
-			return query( fromJS( state ) );
+			return fromJS( state ).set( 'list', state.list );
```

Comment:
Assume that server state is the source of truth, and just use it (instead of transforming with `query` function which deletes existing `list` items).

Since this state tree is a mix of Immutable objects and POJOs, we have to manually reset the `list` as an `Array` from the server-render state. Otherwise it transforms to a `List` and breaks functionality that expects it to be an `Array`.